spec/requests/api/moves_controller_filter_v2_spec.rb

```diff
@@ -0,0 +1,232 @@
+        let(:expected_moves) { Move.where(id: move_without_allocation.id) }
+        let(:unexpected_moves) { Move.where(id: move_with_allocation.id) }
+        let(:filter_value) { false }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+
+      context 'when set to incorrect type' do
+        let(:expected_moves) { Move.where(id: [move_without_allocation.id, move_with_allocation.id]) }
+        let(:unexpected_moves) { Move.none }
+        let(:filter_value) { 'some-value' }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+
+      context 'when set to empty value' do
+        let(:expected_moves) { Move.where(id: [move_without_allocation.id, move_with_allocation.id]) }
+        let(:unexpected_moves) { Move.none }
+        let(:filter_value) { nil }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+
+      context 'when filter not set' do
+        let(:expected_moves) { Move.where(id: [move_without_allocation.id, move_with_allocation.id]) }
+        let(:unexpected_moves) { Move.none }
+        let(:filter_params) { {} }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+    end
+
+    describe 'by cancellation_reason' do
+      let(:filter_params) { { filter: { cancellation_reason: cancellation_reason } } }
+      let(:made_in_error_moves) { create_list :move, 3, :cancelled_made_in_error }
+      let(:supplier_declined_to_move_moves) { create_list :move, 3, :cancelled_supplier_declined_to_move }
+      let(:rejected_moves) { create_list :move, 3, :cancelled_rejected }
+      let(:other_moves) { create_list :move, 3, :cancelled_other }
+
+      context 'when made_in_error' do
+        let(:cancellation_reason) { 'made_in_error' }
+        let(:expected_moves) { made_in_error_moves }
+        let(:unexpected_moves) { supplier_declined_to_move_moves + rejected_moves + other_moves }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+
+      context 'when supplier_declined_to_move' do
+        let(:cancellation_reason) { 'supplier_declined_to_move' }
+        let(:expected_moves) { supplier_declined_to_move_moves }
+        let(:unexpected_moves) { made_in_error_moves + rejected_moves + other_moves }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+
+      context 'when rejected' do
+        let(:cancellation_reason) { 'rejected' }
+        let(:expected_moves) { rejected_moves }
+        let(:unexpected_moves) { made_in_error_moves + supplier_declined_to_move_moves + other_moves }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+
+      context 'when other' do
+        let(:cancellation_reason) { 'other' }
+        let(:expected_moves) { other_moves }
+        let(:unexpected_moves) { made_in_error_moves + supplier_declined_to_move_moves + rejected_moves }
+
+        it_behaves_like 'an api that filters moves correctly'
+      end
+    end
+  end
+end
```

Comment:
Do we need to duplicate this spec? (from `moves_controller_filter_spec.rb`)

Maybe it makes sense to have a filter for `supplier_id` in the V1 too, it does no harm, right?

Reply:
The `Move::Finder` class is shared between both versions of this api so we automatically get the same behaviour for filters/sorting.

I have already written a test for v1 (see line 37 of spec/requests/api/moves_controller_filter_spec.rb)

Reply:
It's confusing because we already supported filtering by supplier id but against the from location suppliers which is what we're moving away from.